### ISSUE TITLE
React native launch debugger  error

### ISSUE THREAD

**Issue by tootwo2** [facebook/nuclide#1410]
### Issue
When i use 'launch debugger' ,there is error info in the console view:
```
SyntaxError: Unexpected token <
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:599:28)
    at Object.Module._extensions..js (module.js:646:10)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)
    at Function.Module._load (module.js:489:3)
    at Function.Module.runMain (module.js:676:10)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3
```

### Versions
* Atom:1.22.1
* Nuclide:0.271.0
* Client OS:macOS 10.12.6
* xcode: 9.2

**Comment by pelmers** [comment]
hi @tootwo2, could you also post your react-native version number? thanks for the report!

**Comment by tootwo2** [comment]
my react-native version is 0.51.0.

**Comment by pelmers** [comment]
thanks, looks like this is known issue with vscode-react-native: https://github.com/Microsoft/vscode-react-native/issues/613

does it happen on every app? I can look into reproducing/fixing the issue

**Comment by tootwo2** [comment]
the problem above is ok.But there is a new error info:
```
Error: Could not start the packager.
    at Q.then.then.conditionResult (/Users/XXX/.atom/packages/nuclide/modules/nuclide-debugger-vsps/VendorLib/vscode-react-native/out/common/node/promise.js:52:23)
    at _fulfilled (/Users/XXX/.atom/packages/nuclide/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/Users/XXX/.atom/packages/nuclide/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/Users/XXX/.atom/packages/nuclide/node_modules/q/q.js:796:13)
    at /Users/XXX/.atom/packages/nuclide/node_modules/q/q.js:556:49
    at runSingle (/Users/XXX/.atom/packages/nuclide/node_modules/q/q.js:137:13)
    at flush (/Users/XXX/.atom/packages/nuclide/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```
@pelmers

**Comment by pelmers** [comment]
that message should appear if the packager has not started 60 seconds after starting the debugger. how do you normally run the packager? you may be successful using the "attach" option to put the debugger on that port.

**Comment by tootwo2** [comment]
how can i change 60 seconds to longer time. @pelmers

**Comment by pelmers** [comment]
hi @tootwo2 , sorry for the late reply: the retry/timeout line should be found at `~/.atom/packages/nuclide/modules/modules/nuclide-debugger-vsps/VendorLib/vscode-react-native/out/common/packager.js` (it should be this line: https://git.io/vbh7m)

By default it checks 30 times, 2 seconds apart. Increasing either value would effectively raise the timeout. If you are still having trouble with higher values, I would recommend starting the packager separately and using the attach method to start the debugger.

For reference, that file is ported from Microsoft's vsocde-react-native debugger tool: https://github.com/Microsoft/vscode-react-native/blob/master/src/common/packager.ts#L286 so you could post a question asking them if you can reproduce from vscode.

**Comment by Unforgiven-wanda** [comment]
I am having the exact same issue
I am not clear on this, is React native debugging in nuclide still supported? Because according to [this](https://stackoverflow.com/a/45538584/2393645) it has been dropped
Also when I launch the debugger from nuclide, Chrome comes up automatically and is the one who gets attached to the process, (see #1389) is there a way to prevent that?
Thank you

**Comment by pelmers** [comment]
@Unforgiven-wanda yes, the react native debugger was dropped in June, but then re-added in release 0.271 by porting from [Microsoft/vscode-react-native](https://github.com/Microsoft/vscode-react-native) through our new vscode debugger protocol support. 

Since then I fixed a couple of issues related to handling source maps (bfd730b9c219bd8258fcc30ec6fcb18cf9255226) and mistakenly expecting a ".vscode" directory to exist (7a4bcdb8ed005f825e26de684c85f6f48ccaf495). (see context at https://github.com/facebook/nuclide/issues/1411). These fixes will appear in the next release.

what version of react native are you running? the currently included debugger only supports <0.50 (fixed in new release by 5cc0055b9cc1055dc75375a21da23ef6a21c6fe7)

**Comment by Unforgiven-wanda** [comment]
> the currently included debugger only supports <0.50
That must be it then. The project I'm currently on runs on the latest stable release of RN 0.51
Thank you for the clarification. Looking forward to this version being supported.

**Comment by tootwo2** [comment]
@pelmers thank you for help
